docker/ubuntu/Dockerfile

```diff
@@ -184,47 +234,38 @@ ENV LDFLAGS ""
 ENV CFLAGS ""
 ENV CXXFLAGS ""
 
-# set SHELL var to avoid /bin/sh fallback in interactive GRASS GIS sessions
-ENV SHELL /bin/bash
-ENV LC_ALL "en_US.UTF-8"
-ENV GRASS_SKIP_MAPSET_OWNER_CHECK 1
-
-# https://proj.org/usage/environmentvars.html#envvar-PROJ_NETWORK
-ENV PROJ_NETWORK=ON
-
-# Create generic GRASS GIS lib name regardless of version number
-RUN ln -sf /usr/local/grass84 /usr/local/grass
-
-# show GRASS GIS, PROJ, GDAL etc versions
-RUN grass --tmp-location EPSG:4326 --exec g.version -rge && \
-    pdal --version && \
-    python3 --version
+# Leave build stage
+FROM common as grass
 
-# Reduce the image size
-RUN apt-get autoremove -y
-RUN apt-get clean -y
-RUN rm -r /src/grass_build/.git
-
-WORKDIR /scripts
+# GRASS GIS specific
+# allow work with MAPSETs that are not owned by current user
+# add GRASS GIS envs for python usage
+ENV GRASSBIN="/usr/local/bin/grass" \
+    GRASS_SKIP_MAPSET_OWNER_CHECK=1 \
+    SHELL="/bin/bash" \
+    # https://proj.org/usage/environmentvars.html#envvar-PROJ_NETWORK
+    PROJ_NETWORK=ON \
+    # GRASSBIN=grass \
+    LC_ALL="en_US.UTF-8" \
+    GISBASE="/usr/local/grass/" \
+    GRASSBIN="/usr/local/bin/grass" \
+    PYTHONPATH="${PYTHONPATH}:/usr/local/grass/etc/python/" \
+    LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/usr/local/grass/lib"
 
-# install external GRASS GIS session Python API
-RUN pip3 install grass-session
+# Copy GRASS GIS from build image
+COPY --from=build /usr/local/bin/grass* /usr/local/bin/
+COPY --from=build /usr/local/grass84 /usr/local/grass84/
+COPY --from=build /src/site-packages /usr/lib/python3.10/
+COPY --from=build /usr/lib/gdalplugins /usr/lib/gdalplugins
```

Comment:
Consider using `COPY --link --from=build` in order to not invalidate the cache below if the files changed. https://docs.docker.com/engine/reference/builder/#copy---link  When using `COPY --link`, the files are copied as an independent scratch layer, and it's content can be changed without affecting the cache of others. It is useful in multi-stage builds when combining files built in independent stages at the end, since the files copied are really independent and no other RUN commands should be unconditionally run below.

Here, there is only the `RUN ln` that we might need to check that the behaviour is still correct.

Reply:
Thanks. Included in 4d4da07

The symbolic link seems to work.